blockchain/blockchain.go

```diff
@@ -1097,7 +1102,8 @@ func (bc *blockchain) commitBlock(blk *block.Block) error {
 		// detach working set so it can be freed by GC
 		blk.WorkingSet = nil
 		if err != nil {
-			return err
+			// TODO: we shouldn't panic once state mutation logic is stable
+			logger.Panic().Err(err).Msg("Error when committing state changes")
```

Comment:
BTW, this is irrelevant, but I still put it here. I change back to Panic if committing states fails. It's easier for debugging to crash instead of continue now when states corrupt.